apps/passport-server/test/devconnect.spec.ts

```diff
@@ -373,6 +381,218 @@ describe("devconnect functionality", function () {
+        db
+      );
+      if (!devconnectPretixAPIConfigFromDB) {
+        throw new Error("Could not load API configuration");
+      }
+
+      const organizer = devconnectPretixAPIConfigFromDB?.organizers[0];
+      const eventConfigID = organizer.events[0].id;
+
+      const ticketsAwaitingSyncBeforeCheckin =
+        await fetchDevconnectTicketsAwaitingSync(db, organizer.orgURL);
+
+      // There are no tickets awaiting sync
+      expect(ticketsAwaitingSyncBeforeCheckin.length).to.eq(0);
+
+      // Grab a ticket and consume it
+      const tickets = await fetchDevconnectPretixTicketsByEvent(
+        db,
+        eventConfigID
+      );
+      const ticket = tickets[0];
+
+      const result = await consumeDevconnectPretixTicket(
+        db,
+        ticket.id,
+        "test@example.com"
+      );
+
+      expect(result).to.be.true;
+
+      const consumedTicket = await fetchDevconnectPretixTicketByTicketId(
+        db,
+        ticket.id
+      );
+
+      // The same ticket should now be consumed
+      expect(consumedTicket?.is_consumed).to.be.true;
+      expect(consumedTicket?.checkin_timestamp).to.be.not.null;
+
+      const ticketsAwaitingSync = await fetchDevconnectTicketsAwaitingSync(
+        db,
+        organizer.orgURL
+      );
+
+      // The consumed ticket is awaiting sync
+      expect(ticketsAwaitingSync.length).to.eq(1);
+      expect(ticketsAwaitingSync[0]).to.deep.include(consumedTicket);
+
+      // Set up a sync manager for a single organizer
+      const os = new OrganizerSync(
+        organizer,
+        new DevconnectPretixAPI({ requestsPerInterval: 300 }),
+        application.services.rollbarService,
+        application.context.dbPool
+      );
+
+      let receivedCheckIn = false;
+
+      server.use(
+        rest.post(
+          organizer.orgURL + "/checkinrpc/redeem",
+          async (req, res, ctx) => {
+            const body = new Map(Object.entries(await req.json()));
+            if (
+              !body.has("secret") ||
+              !body.has("lists") ||
+              typeof body.get("secret") !== "string" ||
+              !Array.isArray(body.get("lists"))
+            ) {
+              return res(ctx.status(400), ctx.json({}));
+            }
+
+            receivedCheckIn = true;
+            return res(ctx.json({ status: "ok" }));
+          }
+        )
+      );
+
+      // Because we're not patching the data from Pretix, tickets will now
+      // have no check-ins
+      expect(await os.run()).to.not.throw;
+
+      // The Pretix backend should have received a check-in request
+      expect(receivedCheckIn).to.be.true;
+
+      const consumedTicketAfterSync =
+        await fetchDevconnectPretixTicketByTicketId(db, ticket.id);
+
+      // Ticket should be marked as consumed
+      expect(consumedTicketAfterSync?.is_consumed).to.be.true;
+      // And ticket should have a pretix checkin timestamp!
+      expect(consumedTicketAfterSync?.pretix_checkin_timestamp).to.be.not.null;
```

Comment:
maybe add an assertion that the checker is not `"Pretix"`, but rather the checker?